orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/DeployManifestTask.java

```diff
@@ -70,25 +72,25 @@ public TaskResult execute(@Nonnull Stage stage) {
     Map<String, Object> task = new HashMap<>(context);
     String artifactSource = context.getSource();
     if (StringUtils.isNotEmpty(artifactSource) && artifactSource.equals("artifact")) {
-      String manifestArtifactId = context.getManifestArtifactId();
-      if (manifestArtifactId == null) {
+      Artifact manifestArtifact = artifactResolver.getBoundArtifactForStage(stage, context.getManifestArtifactId(),
+        context.getManifestArtifact());
+
+      if (manifestArtifact == null) {
         throw new IllegalArgumentException("No manifest artifact was specified.");
       }
 
-      String manifestArtifactAccount = context.getManifestArtifactAccount();
-      if (manifestArtifactAccount == null) {
-        throw new IllegalArgumentException("No manifest artifact account was specified.");
+      // Once the legacy artifacts feature is removed, all trigger expected artifacts will be required to define
+      // an account up front.
+      if(context.getManifestArtifactId() != null && manifestArtifact.getArtifactAccount() == null) {
```

Comment:
Should we be checking `context.getManifestArtifactAccount() != null` here?  I suppose that we expect `context.getManifestArtifactId()` to be `null` iff `context.getManifestArtifactAccount()` is `null` but it might be clearer to check the relevant field.

Reply:
I'm afraid of a dangling `manifestArtifactAcount` as folks migrate pipelines. So trying to be extra safe.

Reply:
Ok, makes sense :+1:

Reply:
Good point. Updated.